### ISSUE TITLE
Cloud images should be usable also after a new release

### ISSUE THREAD

**Issue by danfruehauf** [CentOS/sig-cloud-instance-build#126]
When using cloud images (specifically Vagrant/VirtualBox images), older image releases will lack packages in remote repositories, rendering those images useless for some.

A concrete example is image version 1803.01 on the day of writing this ticket. The image ships with kernel-3.10.0-693.11.6, yet kernel-* packages with the corresponding versions are nowhere to be found (like kernel-devel-3.10.0-693.11.6). Those packages are also not in the vault repositories.

An update+reboot can solve the problem, however part of the idea of having a cloud image is to not have to update+reboot it as the first operation.

**Comment by danfruehauf** [comment]
@kbsingh Please modify/add more information as you see fit. We have discussed it more thoroughly on #centos-devel.

@lpancescu FYI

**Comment by danfruehauf** [comment]
> The VirtualBox Guest Additions are not necessary for our Vagrant images and they bring a number of problems:

I understand that my specific scenario is with vagrant-vbguest. However I can phrase it differently so it doesn't include vagrant-vbguest:
*I boot an image and I have no kernel-devel package available for it* - that's the issue at hand.

When I use a ready made image, I do not expect the first operation on it to be an update and a reboot. I understand that it is recommended, but it must not be necessary.

**Comment by yeroc** [comment]
I seem to have just been bitten by this issue.  Any progress on this?

**Comment by lpancescu** [comment]
> I seem to have just been bitten by this issue. Any progress on this?

No, and there's nothing I can do about this on the image side. When a new point release appears, the previous ones are no longer supported and will be moved to vault. We can try to speed up producing the Vagrant images after a point release, but images you already have will always have this problem when creating a new instance. I don't really think the CentOS core team would be willing to change how the archive is working just to appease a bug in vagrant-vbguest, which its authors could probably change to try vault if they don't find the package in the normal archive.

I think your best chances are either to convince vagrant-vbguest to enhance their implementation, use my starter kit linked above, or just don't use the VirtualBox Guest Additions, which aren't necessary (I've never used them, they just slow down kernel updates with no practical benefit).

**Comment by yeroc** [comment]
Understood.  It's truly unfortunate the core team have taken that stance with how the vault/archiving process works.  More and more as issues like this crop up I get feedback from teammates and others that tell me other distributions (such as Ubuntu or even Fedora) are more reliable so these things are doing real harm to the CentOS "brand".  Ironically, it feels like this has been getting worse since RedHat increased involvement/funding in the product.  

That said, which is the appropriate project to raise this issue under?

**Comment by lpancescu** [comment]
@yeroc said:
> It's truly unfortunate the core team have taken that stance with how the vault/archiving process works.

I'm not a part of the core team, but I think moving old packages into vault makes sense: there is no support for older point releases, neither assistance on the mailing lists or IRC (you'll be told to upgrade first), nor, most importantly, security updates. Having vulnerable packages in the main archive is not what anybody wants, because someone might end up installing them. vagrant-vbguest is in the best position to make this work, I would try contacting its authors if I were you.

As a matter of policy, I will not engage in any discussion regarding the merits of different Linux distributions. If you think something else serves your needs better, that's your personal choice and I respect that. Besides, this is an issue tracker, not a forum for such discussions.
